program/src/error.rs

```diff
@@ -153,6 +153,22 @@ pub enum RuleSetError {
     /// 35 - Program Owned Set check failed
     #[error("Program Owned Set check failed")]
     ProgramOwnedSetCheckFailed,
+
+    /// 36 - Invalid compare operator
+    #[error("Invalid compare operator")]
+    InvalidCompareOp,
+
+    /// 37 - Invalid constraint type value
+    #[error("Invalid constraint type value")]
+    InvalidConstraintType,
+
+    /// 38 - Deserialization error
+    #[error("Deserialization error")]
+    DeserializationError,
```

Comment:
This is a good point. The `DeserializationError` is used in a context where it can either be triggered by RMP or bytemuch. I renamed it to `RuleSetReadFailed`, so it is more generic. When there are issues with the alignment, there is already logs that provide more information – e.g.: 

```
// sanity check: make sure we got the correct slice size
if (slice_end + 1) > bytes.len() {
        msg!("Invalid slice end: {} > {}", slice_end, bytes.len());
        return Err(RuleSetError::RuleSetReadFailed);
}
```